tests/test_install.py

```diff
@@ -27,510 +26,159 @@ def generate_random_path():
-                "environments and packages."
-            ) in output
-        except:
-            raise
-        finally:
-            chdir(cwd)
-
-
-class FileTests(unittest.TestCase):
-    def setUp(self):
-        self.tmpdir = tempfile.mkdtemp()
-        self.tmpfname = join(self.tmpdir, "testfile")
-
-    def tearDown(self):
-        shutil.rmtree(self.tmpdir)
-
-    def test_default_text(self):
-        with open(self.tmpfname, "w") as fo:
-            fo.write("#!/opt/anaconda1anaconda2anaconda3/bin/python\n" 'echo "Hello"\n')
-        update_prefix(self.tmpfname, "/usr/local")
-        with open(self.tmpfname) as fi:
-            data = fi.read()
-            self.assertEqual(data, "#!/usr/local/bin/python\n" 'echo "Hello"\n')
-
-    @pytest.mark.skipif(on_win, reason="test is invalid on windows")
-    def test_long_default_text(self):
-        with open(self.tmpfname, "w") as fo:
-            fo.write(
-                "#!/opt/anaconda1anaconda2anaconda3/bin/python -O\n" 'echo "Hello"\n'
-            )
-        new_prefix = "/usr/local/{}".format("1234567890" * 52)
-        update_prefix(self.tmpfname, new_prefix)
-        with open(self.tmpfname) as fi:
-            data = fi.read()
-            self.assertEqual(data, "#!/usr/bin/env python -O\n" 'echo "Hello"\n')
-
-    @pytest.mark.skipif(on_win, reason="no binary replacement done on win")
-    def test_binary(self):
-        with open(self.tmpfname, "wb") as fo:
-            fo.write(b"\x7fELF.../some-placeholder/lib/libfoo.so\0")
-        update_prefix(
-            self.tmpfname,
-            "/usr/local",
-            placeholder="/some-placeholder",
-            mode=FileMode.binary,
+        # this is the actual test: change the embedded prefix and make sure that the exe runs.
+        data = open("conda.exe", "rb").read()
+        fixed_data = binary_replace(data, original_prefix, sys.executable)
+        with open("conda.fixed.exe", "wb") as f:
+            f.write(fixed_data)
+        # without a valid shebang in the exe, this should fail
+        with pytest.raises(subprocess.CalledProcessError):
+            subprocess.check_call(["conda.exe", "-h"])
+
+        process = subprocess.Popen(
+            ["conda.fixed.exe", "-h"],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
         )
-        with open(self.tmpfname, "rb") as fi:
-            data = fi.read()
-            self.assertEqual(
-                data, b"\x7fELF.../usr/local/lib/libfoo.so\0\0\0\0\0\0\0\0"
-            )
-
-    def test_trash_outside_prefix(self):
-        tmp_dir = tempfile.mkdtemp()
-        rel = relpath(tmp_dir, context.root_prefix)
-        self.assertTrue(rel.startswith(".."))
-        move_path_to_trash(tmp_dir)
-        self.assertFalse(exists(tmp_dir))
-        makedirs(tmp_dir)
-        move_path_to_trash(tmp_dir)
-        self.assertFalse(exists(tmp_dir))
-
-
-# class remove_readonly_TestCase(unittest.TestCase):
```

Comment:
Huh, what's this all about?

Reply:
tests that should've been previously removed (not commented) https://github.com/conda/conda/commit/eaf04ae157d40a4a6f03b9bc160addd8c816f93d